### PR TITLE
Add linalg.lu_solve

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #73878
* #73877
* #74046
* #77637
* #77636
* #77635
* __->__ #77634

Relanding https://github.com/pytorch/pytorch/pull/72935

Differential Revision: [D36793144](https://our.internmc.facebook.com/intern/diff/D36793144)

### CODE REVIEW

**Comment by ngimel** [comment]
Do we need to run this on ci/trunk? Why was it reverted previously?

**Comment by Lezcano** [comment]
It broke an internal build. See https://github.com/pytorch/pytorch/pull/72935#issuecomment-1121459787 https://github.com/pytorch/pytorch/pull/72935#issuecomment-1121470502 https://github.com/pytorch/pytorch/pull/72935#issuecomment-1121473646 https://github.com/pytorch/pytorch/pull/72935#issuecomment-1128109367

**Comment by ngimel** [comment]
Ok, so what next? We can't just reland it because it'll break internal build again?

**Comment by Lezcano** [comment]
According to https://github.com/pytorch/pytorch/pull/72935#issuecomment-1128109367, I'd assume that @malfet already corrected the internal error?

**Comment by ngimel** [comment]
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36793144).

**Comment by ngimel** [comment]
I'm still getting internal build failures on this one 
```
Summary: 
ld.lld: error: undefined symbol: at::native::DispatchStubImpl::get_call_ptr(c10::DeviceType, void*)
stderr: ld.lld: error: undefined symbol: at::native::DispatchStubImpl::get_call_ptr(c10::DeviceType, void*)
``` 
so we cannot land it.

**Comment by malfet** [comment]
> @malfet any updates on this one?

Sorry, completely slipped my mind. Do you mind rebasing it against latest viable/strict and let me try the import again...

**Comment by Lezcano** [comment]
@malfet rebased.
I needed to add an import that was missing to a file that was unrelated to this PR (idk why but it was not building locally...)

**Comment by malfet** [comment]
Ok, I finally understand what is going on(torch_cuda in ideal world should be oblivious of supported CPU architectures, which is enforced internally but not in OSS build, so have an idea how to fix it internally):
https://github.com/pytorch/pytorch/blob/32461ed319866f06c95c1bedd61a6acbceca9529/aten/src/ATen/native/DispatchStub.h#L75-L80

**Review by malfet — APPROVED** [review]

File: aten/src/ATen/native/cuda/linalg/BatchLinearAlgebra.cpp
Comment (kit1980) [inline aten/src/ATen/native/cuda/linalg/BatchLinearAlgebra.cpp:3060]: @lezcano is it time to remove this now?

Comment (lezcano) [reply aten/src/ATen/native/cuda/linalg/BatchLinearAlgebra.cpp:3060]: It's rather minor, but sure, happy to approve a PR that removes this provided that CI passes

**Pull request closed without merging**
